Sources/Ledger/PartialTxOut.swift

```diff
@@ -48,17 +55,22 @@ extension PartialTxOut {
               let publicKey = RistrettoPublic(txOutRecord.txOutPublicKeyData),
               let commitment = TxOutUtils.reconstructCommitment(
                                                     maskedValue: txOutRecord.txOutAmountMaskedValue,
+                                                    maskedTokenId: txOutRecord.txOutAmountMaskedTokenID,
                                                     publicKey: publicKey,
                                                     viewPrivateKey: viewKey),
               Self.isCrc32Matching(commitment, txOutRecord: txOutRecord)
         else {
             return nil
         }
-
+        guard [0,4,8].contains(txOutRecord.txOutAmountMaskedTokenID.count) else {
```

Comment:
why not just add this to the long list of prior guard statements?  (like the Self.isCrc32Matching check)?

Reply:
ya not sure, ill move it up